packages/types/src/utils.ts

```diff
@@ -1,17 +1,25 @@
-import { PaymentTypes } from '.';
+import { ExtensionTypes } from '.';
 import {
   ICreationParameters,
   IOriginalRequestCreationParameters,
   ISubsequentRequestCreationParameters,
 } from './extensions/pn-any-stream-reference-based-types';
 
+/*
+ * Helper to properly type keys of an enum with string values.
+ * See: https://www.petermorlion.com/iterating-a-typescript-enum/
+ */
+function enumKeys<O extends Record<string, unknown>, K extends keyof O = keyof O>(obj: O): K[] {
+  return Object.keys(obj).filter((k) => Number.isNaN(+k)) as K[];
+}
+
 /**
  * Types a value like ExtensionType into a paymentNetworkID enum element if possible
- * @param value Example: ExtensionTypes.ID.PAYMENT_NETWORK_ERC20_PROXY_CONTRACT
+ * @param value Example: ExtensionTypes.PAYMENT_NETWORK_ID.ERC20_PROXY_CONTRACT
  */
-export function isPaymentNetworkId(value: any): value is PaymentTypes.PAYMENT_NETWORK_ID {
-  for (const pn in PaymentTypes.PAYMENT_NETWORK_ID) {
-    if (PaymentTypes.PAYMENT_NETWORK_ID[pn] === value) {
+export function isPaymentNetworkId(value: any): value is ExtensionTypes.PAYMENT_NETWORK_ID {
+  for (const pn of enumKeys(ExtensionTypes.PAYMENT_NETWORK_ID)) {
```

Comment:
The old trick was not working anymore

Reply:
I wonder if `isPaymentNetworkId` is still necessary. It seems to be there to account for the possibility of the ID not being a payment network ID, but now we have a strong type to enforce it? 

btw, you could also do `Object.values(ExtensionTypes.PAYMENT_NETWORK_ID).includes(value)`

Reply:
I agree, it should probably be removed or at least accept only `ID` values.